### PR TITLE
Create best-practices.md

### PR DESCRIPTION
**Pull request by jasonblais**
Waiting on https://github.com/mattermost/mattermost-developer-documentation/pull/297 and https://github.com/mattermost/mattermost-developer-documentation/pull/295

### CODE REVIEW

File: site/content/extend/plugins/best-practices.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+   ```
+   git checkout --orphan dummy-master #create the dummy master branch
+   git rm -rf . #remove everything from this branch
+   git push origin dummy-master
+   ```
+   
+3. Create a dummy-feature branch from dummy-master:
+
+   ```
+   git checkout -b dummy-feature
```
Comment (hanzei) [inline site/content/extend/plugins/best-practices.md]: Maybe call it `review` instead?

File: site/content/extend/plugins/best-practices.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+## How can I review the entire code base of a plugin?
+
+Carry out the following steps:
+
+1. Take a backup of project directory
+2. Create a dummy-master branch with no code:
+
+   ```
+   git checkout --orphan dummy-master #create the dummy master branch
```
Comment (hanzei) [inline site/content/extend/plugins/best-practices.md]: 2/5 to not add comments in the same line as commands. This way it's not easy copy&past-able.

File: site/content/extend/plugins/best-practices.md (hunk 1 above)
Comment (jasonblais) [reply site/content/extend/plugins/best-practices.md]: You mean `dummy-review`?

Comment (hanzei) [reply site/content/extend/plugins/best-practices.md]: No, just `review`. The whole point of this process is the review the plugin code. Hence, `review` seams to be the best branch name to me.

Comment (jasonblais) [reply site/content/extend/plugins/best-practices.md]: `dummy-XXX` helps make it clear all the review work is done against a dummy branch and hence changes won't affect master. 1/5

Comment (hanzei) [reply site/content/extend/plugins/best-practices.md]: If you are 1/5, I'm fine with `dummy-review`.

**Comment by jasonblais** [comment]
Thanks @hanzei, PR updated per feedback

File: site/content/extend/plugins/best-practices.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+---
+
+See here for [server-specific best practices for plugins](/extend/plugins/server/best-practices/). Webapp-specific best practices are incoming.
+
+## How can I review the entire code base of a plugin?
+
+Carry out the following steps:
+
+1. Take a backup of project directory
+2. Create a dummy-master branch with no code:
```
Comment (cpoile) [inline site/content/extend/plugins/best-practices.md]: What I did was commit the code that I didn't want to review. Like, you probably don't want to review boilerplate files and boilerplate code.
So after the `git rm -rf` you could have `git add ....` where `....` are the files that you don't want to review.
then `git commit -m "Files we don't want to review"`
Something like that.

File: site/content/extend/plugins/best-practices.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+   git rm -rf
+   git push origin dummy-master
+   ```
+   
+3. Create a dummy-review branch from dummy-master:
+
+   ```
+   git checkout -b dummy-review
+   git add
+   git commit -m "Full checkin"
```
Comment (cpoile) [inline site/content/extend/plugins/best-practices.md]:
```suggestion
   git commit -m "Files to review"
```

File: site/content/extend/plugins/best-practices.md (hunk 3 above)
Comment (jasonblais) [reply site/content/extend/plugins/best-practices.md]: @cpoile How's this?

----

## How can I review the entire code base of a plugin?

Carry out the following steps:

1. Take a backup of project directory
2. Create a dummy-master branch with no code:

   ```
   git checkout --orphan dummy-master
   git rm -rf
   ```

3. Commit code that you do not want to review:

   ```
   git add <files excluded from review>
   git push origin dummy-master
   ```
   
4. Create a dummy-review branch from dummy-master:

   ```
   git checkout -b dummy-review
   git add
   git commit -m "Files to review"
   git push origin dummy-review
   ```
   
5. Create a PR from dummy-review -> dummy-master

6. Code review on the resulting PR

Comment (cpoile) [reply site/content/extend/plugins/best-practices.md]: @jasonblais Awesome!

Comment (cpoile) [reply site/content/extend/plugins/best-practices.md]: @jasonblais oh wait, you might want to gut a commit in step 3

Comment (jasonblais) [reply site/content/extend/plugins/best-practices.md]: @cpoile Updated 👍

**Comment by jasonblais** [comment]
I've updated so that it's no longer dependent on #295, merging

**Pull request merged**